### DISCUSSION TITLE
Fix react/client hydration handling of node.class

### DISCUSSION THREAD

**Discussion by eigood** [withastro/roadmap#917] [Proposal]
### Body

https://github.com/withastro/astro/blob/e39ee5662d870a91a75b6b3ac441b9e3ef9e1063/packages/integrations/react/client.js#L16

https://github.com/withastro/astro/blob/e39ee5662d870a91a75b6b3ac441b9e3ef9e1063/packages/integrations/react/vnode-children.js#L19

Vnode-children handles mapping of class->className, but this only runs server-side.  When hydrating to the client, and converting the raw browser DOM back to vnodes, the same mapping does not take place.  Additionally, arrays of children are not getting a proper "key" property, so there is a warning getting printed.

Fyi, I have experimentalReactChildren=true

**Comment by florian-lefebvre** [comment]
Hey that sounds like a bug to me! I may be wrong tho as I've not contributed to this part of Astro but I think you can open an issue directly at https://github.com/withastro/astro/issues

**Reply by eigood** [reply]
Sure, but there is a lot of boilerplate needed to file a bug.
